Source/RunActivity/Viewer3D/RollingStock/MSTSLocomotiveViewer.cs

```diff
@@ -2134,29 +2134,15 @@ public virtual int GetDrawIndex()
                         Locomotive.DynamicBrakePercent : Locomotive.LocalDynamicBrakePercent;
                     if (Locomotive.DynamicBrakeController != null)
                     {
-                        if (dynBrakePercent == -1)
+                        if (dynBrakePercent <= 0)
                         {
                             index = 0;
                             break;
                         }
                         if (!Locomotive.HasSmoothStruc)
-                        {
                             index = Locomotive.DynamicBrakeController != null ? Locomotive.DynamicBrakeController.CurrentNotch : 0;
-                        }
                         else
-                        {
-                            if (Locomotive.CruiseControl != null)
-                            {
-                                if ((Locomotive.CruiseControl.SpeedRegMode == Simulation.RollingStocks.SubSystems.CruiseControl.SpeedRegulatorMode.Auto && !Locomotive.CruiseControl.DynamicBrakePriority) || Locomotive.DynamicBrakeIntervention > 0)
-                                {
-                                    index = 0;
-                                }
-                                else
-                                    index = PercentToIndex(dynBrakePercent);
-                            }
-                            else
-                                index = PercentToIndex(dynBrakePercent);
-                        }
+                            index = PercentToIndex(dynBrakePercent);
```

Comment:
```suggestion
                            index = PercentToIndex(Locomotive.DynamicBrakeController.CurrentValue);
```

Reply:
This whole area was a bit of a mess when it came to null checks on DynamicBrakeController. Cleaned up in the next version.

Reply:
My point was something else. I think the current behaviour is not doing what it should for smooth controllers. The right code should be:
```csharp
if (DynamicBrakeController != null)
{
    if (HasSmoothStruct) index = PercentToIndex(DynamicBrakeController.CurrentValue);
    else index = DynamicBrakeController.CurrentNotch;
}
else
{
    index = PercentToIndex(dynBrakePercent);
}
```

That is, we should use the controller also for smooth controllers. Otherwise, if blending is applied the lever will show the blending value, which is incorrect.

Reply:
Seems like blended braking doesn't affect the LocalDynamicBrakePercent, so that's accounted for...maybe not foolproof though. However, I did notice combined controllers were worse and would show the first frame of dynamic braking during blended braking. Mitigated that as well.